apps/namadillo/src/atoms/chain/atoms.ts

```diff
@@ -2,36 +2,26 @@ import namada from "@namada/chains/chains/namada";
 import { integrations } from "@namada/integrations";
 import { indexerApiAtom } from "atoms/api";
 import {
-  defaultServerConfigAtom,
   indexerUrlAtom,
   namadaExtensionConnectedAtom,
   rpcUrlAtom,
 } from "atoms/settings";
 import { queryDependentFn } from "atoms/utils";
+import { atom } from "jotai";
 import { atomWithQuery } from "jotai-tanstack-query";
 import { ChainParameters, ChainSettings } from "types";
 import { fetchChainParameters, fetchRpcUrlFromIndexer } from "./services";
 
 export const chainAtom = atomWithQuery<ChainSettings>((get) => {
   const chainParameters = get(chainParametersAtom);
-  const indexerRpcUrl = get(indexerRpcUrlAtom);
   const rpcUrl = get(rpcUrlAtom);
-  const indexerUrl = get(indexerUrlAtom);
-  const tomlConfig = get(defaultServerConfigAtom);
   const extensionConnected = get(namadaExtensionConnectedAtom);
 
   return {
-    queryKey: [
-      "chain",
-      rpcUrl,
-      chainParameters,
-      indexerRpcUrl,
-      extensionConnected,
-    ],
+    queryKey: ["chain", rpcUrl, chainParameters.data, extensionConnected],
```

Comment:
Passing only the `.data` because this is what we really use by the `queryFn`. Also, removed the other values as the comment below